_studio/mfx_lib/vpp/src/mfx_vpp_sw_core.cpp

```diff
@@ -426,8 +426,30 @@ mfxStatus VideoVPPBase::QueryIOSurf(VideoCORE* core, mfxVideoParam *par, mfxFram
 
         mfxSts = CheckIOPattern_AndSetIOMemTypes(par->IOPattern, &(request[VPP_IN].Type), &(request[VPP_OUT].Type), bSWLib);
         MFX_CHECK_STS(mfxSts);
+#if defined (MFX_ENABLE_HW_ONLY_VPP)
         return (bSWLib)? MFX_ERR_UNSUPPORTED : MFX_ERR_NONE;
+#else
+        return (bSWLib)? MFX_WRN_PARTIAL_ACCELERATION : MFX_ERR_NONE;
     }
+    else
+    {
```

Comment:
do you really need this 'else' part? and if yes - why?

Reply:
The else if for the case where core->GetPlatformType() does not return  MFX_PLATFORM_HARDWARE.  
I suspect that the wrong piece of code (corresponding to camera plugin) had been copied to the open source code.  It assumes that only HW VPP is supported.  which is not true for MCTF.
I I'll remove the code corresponding to camera plugin as it is not in open source.

Reply:
Yes, this looks to be so. From the scripts stripping closed source version:
```
# fgrep -rsn AS_CAMERA_PLUGIN  builder/
builder/open_source_symbols_undefine.list:16:AS_CAMERA_PLUGIN
# fgrep -rsn MFX_ENABLE_HW_ONLY_VPP builder/
builder/open_source_symbols_define.list:6:MFX_ENABLE_HW_ONLY_VPP
```
@onabiull: will a strip script left both macro paths for MFX_ENABLE_HW_ONLY_VPP or not under the above conditions?

I don't think that you use this 'else' path for MCTF... Actually I am not sure whether we have any use cases for that... Oleg, how do you think - do we need this 'else' at all in open source?